app/src/ui/drag-elements/commit-drag-element.tsx

```diff
@@ -26,60 +29,88 @@ export class CommitDragElement extends React.Component<
   public constructor(props: ICommitDragElementProps) {
     super(props)
     this.state = {
-      branchName: null,
+      showTooltip: false,
+      currentDropTarget: null,
     }
 
-    dragAndDropManager.onEnterDropTarget(branchName => {
-      this.setBranchName(branchName)
+    dragAndDropManager.onEnterDropTarget(dropTarget => {
+      this.setState({ currentDropTarget: dropTarget })
+      switch (dropTarget.type) {
+        case DropTargetType.Branch:
+        case DropTargetType.Commit:
+          this.setToolTipTimer(1500)
+          break
+        default:
+          assertNever(dropTarget, `Unknown drop target type: ${dropTarget}`)
+      }
     })
 
     dragAndDropManager.onLeaveDropTarget(() => {
-      this.setState({ branchName: null })
+      this.setState({ currentDropTarget: null, showTooltip: false })
     })
   }
 
-  private setBranchName(branchName: string) {
+  private setToolTipTimer(time: number) {
     if (__DARWIN__) {
       // For macOs, we styled the copy to message to look like a native tool tip
       // that appears when hovering over a element with the title attribute. We
       // also are implementing this timeout to have similar hover-to-see feel.
-      this.setState({ branchName: null })
+      this.setState({ showTooltip: false })
 
       if (this.timeoutId !== null) {
         window.clearTimeout(this.timeoutId)
       }
 
       this.timeoutId = window.setTimeout(
-        () => this.setState({ branchName }),
-        1500
+        () => this.setState({ showTooltip: true }),
+        time
       )
     } else {
-      this.setState({ branchName })
+      this.setState({ showTooltip: true })
     }
   }
 
-  /**
-   * The "copy to" label is a windows convention we are implementing to provide
-   * a more intuitive ux for windows users.
-   */
-  private renderDragCopyLabel(count: number) {
-    const { branchName } = this.state
-    if (branchName === null) {
+  private renderDragToolTip() {
+    const { showTooltip, currentDropTarget } = this.state
+    if (!showTooltip || currentDropTarget === null) {
       return
     }
 
-    const copyToPlus = __DARWIN__ ? null : (
-      <Octicon symbol={OcticonSymbol.plus} />
-    )
+    let toolTipContents
+    switch (currentDropTarget.type) {
+      case DropTargetType.Branch:
+        const copyToPlus = __DARWIN__ ? null : (
+          <Octicon className="copy-to-icon" symbol={OcticonSymbol.plus} />
+        )
+        toolTipContents = (
+          <>
+            {copyToPlus}
+            <span>
+              <span className="copy-to">Copy to </span>
+              <span className="branch-name">
+                {currentDropTarget.branchName}
+              </span>
+            </span>
+          </>
+        )
+        break
+      case DropTargetType.Commit:
+        toolTipContents = (
+          <>
+            <span>Squash {this.props.selectedCommits.length + 1} commits</span>
```

Comment:
I know it sounds dumb, but maybe worth adding a comment about that `+ 1` for people lacking context (or ourselves when we forget about it 😂 )